riscv-plic.adoc

```diff
@@ -17,102 +17,592 @@ https://creativecommons.org/licenses/by/4.0/.
+|===
+| *Device Tree Syntax* | *Description* | *Property Type* | *Property Value*
+|ndev
+|Number of devices connected to this PLIC
+|<u32>
+|Platform-specific
+
+|int-context
+|Interrupt Operation Parameters for the Hart Contexts
+|<prop-encoded-array> +
+This property describes the interrupt sources on specific RISC-V Hart context.
+Multiple `int-context` properties could be declared in PLICv1 Device Tree. If multiple
+`int-context` are declared, the secquence of `int-context` must be in the same order
+of supported contexts organized in Interuupt Enable block.
+Platform vendor shall spec-out the platform-implementation in platform specification. +
+ +
+*parent-cpu* _<phandle>_ +
+The parent CPU node. +
+ +
+*hart-id:* _<u32>_ +
+The Hart ID of this CPU. +
+ +
+*privilege-mode:* _<string>_ +
+The privilege mode of this context. The value could be one of below, +
+  "M-Mode", +
+  "S-Mode", +
+  "U-Mode" +
+ +
+|Platform-specific
+
+|===
+			
+### For the platforms the OS image with device driver which well knows the configuration of RISC-V PLICv1:
+
+In this use case, the Device Tree of RISC-V PLICv1 could be small and simple.
+
+        interrupt-controller@c000000 {
+　　　　　　　　　#interrupt-cells = <0x00000001>;
+　　　　　　　　　compatible = "riscv,plicv1", "sifive, plic";
+　　　　　　　　　interrupt-controller;
+　　　　　　　　　interrupts-extended = <0x00000010 0x0000000b 0x00000011 0x0000000b 0x00000011 0x00000009 0x00000012 0x0000000b 0x00000012 0x00000009 0x00000013 0x0000000b 0x00000013 0x00000009 0x00000014 0x0000000b 0x00000014 0x00000009>;	
+　　　　　　　　　reg = <0x00000000 0x0c000000 0x00000000 0x04000000>;
+　　　　　　　　　reg-names = "control";
+        };
+	
+		
+### For the single OS image with the device driver supports RISC-V PLICv1-Compliant PLICs:
+
+In this use case, the Device Tree of RISC-V PLIC is more complex in order to support single OS image and device driver. Below Device Tree takes SiFive PLIC as reference,
+
+        interrupt-controller@c000000 {
+　　　　　　　　　#interrupt-cells = <0x00000001>;
+　　　　　　　　　compatible = "riscv,plicv1";
+　　　　　　　　　interrupt-controller;
+　　　　　　　　　interrupts-extended = <0x00000010 0x0000000b 0x00000011 0x0000000b 0x00000011 0x00000009 0x00000012 0x0000000b 0x00000012 0x00000009 0x00000013 0x0000000b 0x00000013 0x00000009 0x00000014 0x0000000b 0x00000014 0x00000009>;	
+　　　　　　　　　reg = <0x00000000 0x0c000000 0x00000000 0x04000000>;
+　　　　　　　　　reg-names = "control";
+　　　　　　　　　riscv,ndev = <0x00000035>;			
+　　　　　　　　　riscv,int-context = <&cpu0 0x00000000 "M-Mode">;
+　　　　　　　　　riscv,int-context = <&cpu0 0x00000001 "M-Mode">; 
+　　　　　　　　　riscv,int-context = <&cpu0 0x00000001 "S-Mode">;
+　　　　　　　　　riscv,int-context = <&cpu0 0x00000002 "M-Mode">; 
+　　　　　　　　　riscv,int-context = <&cpu0 0x00000002 "S-Mode">; 
+　　　　　　　　　riscv,int-context = <&cpu0 0x00000003 "M-Mode">;
+　　　　　　　　　riscv,int-context = <&cpu0 0x00000003 "S-Mode">; 
+　　　　　　　　　riscv,int-context = <&cpu0 0x00000004 "M-Mode">; 
+　　　　　　　　　riscv,int-context = <&cpu0 0x00000004 "S-Mode">; 			
```

Comment:
These lines are redundant to the `interupts-extended` attribute. Specifically, that attribute is formatted as pairs of cells where the first is the phandle of the hart (for which there is a mapping to hartid elsewhere in the device tree), and the second is the interrupt number (0xb=M-mode external interrupt, 0x9=S-mode external interrupt).

Reply:
oh I see. I don't have ideas about what interrupt-extended means. thanks for reviewing this.